### PR TITLE
SONARIAC-1345 Highlight precise simple value in Helm expression in primary location

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: iac-common/src/main/java/org/sonar/iac/common/api/tree/impl/TextRange.java [hunk 1]
```diff
@@ -60,4 +64,33 @@ public int hashCode() {
   public String toString() {
     return "[" + start.line() + ":" + start.lineOffset() + "/" + end.line() + ":" + end.lineOffset() + "]";
   }
+
+  public TextRange trimToText(String content) {
```
Comment (petertrr) [inline iac-common/src/main/java/org/sonar/iac/common/api/tree/impl/TextRange.java:68]: Maybe `trimEndToText`? It was not obvious to me, what this method does, by its name

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/helm/tree/utils/GoTemplateAstHelper.java [hunk 2]
```diff
@@ -47,7 +48,11 @@ public static List<ValuePath> findNodes(GoTemplateTree tree, TextRange range, St
...
       .filter(hasOverlayingLocation(location))
-      .map(FieldNode.class::cast)
+      .map(FieldNode.class::cast);
+  }
+
+  public static List<ValuePath> findValuePaths(GoTemplateTree tree, TextRange range, String sourceText) {
+    return findValuePathNodes(tree, range, sourceText)
       .map(FieldNode::identifiers)
       .map(ValuePath::new)
       .collect(Collectors.toList());
```
Comment (petertrr) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/helm/tree/utils/GoTemplateAstHelper.java:58]:
```suggestion
      .toList();
```
Java 17 :)

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitor.java [hunk 3]
```diff
@@ -92,6 +103,31 @@ protected void reportIssue(@Nullable TextRange textRange, String message, List<S
       currentCtx.reportIssue(ruleKey, shiftedTextRange, message, enhancedAndAdjustedSecondaryLocations);
     }
 
+    private TextRange convertToHelmValuePathTextRange(TextRange shiftedTextRange) {
+      if (currentCtx instanceof HelmInputFileContext) {
```
Comment (petertrr) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitor.java:107]:
```suggestion
      if (currentCtx instanceof HelmInputFileContext helmCtx) {
```
Java 17 :)

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitor.java [hunk 4]
```diff
@@ -92,6 +103,31 @@ protected void reportIssue(@Nullable TextRange textRange, String message, List<S
...
+          try {
+            var contents = currentCtx.inputFile.contents();
+            // The go template tree contains locations aligned to source code with additional trailing line numbers comments
+            var valuePathNodes = GoTemplateAstHelper.findValuePathNodes(goTemplateTree, shiftedTextRange, sourceWithComments);
+            var textRanges = valuePathNodes.map(FieldNode::location)
+              .map(location -> location.toTextRange(sourceWithComments))
+              .collect(Collectors.toList());
+            if (!textRanges.isEmpty()) {
+              // The text range may be too big, so it needs to be adjusted to the original source code
+              //TODO: When SONARIAC-1337 wil be implemented maybe this will be not needed anymore.
```
Comment (petertrr) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitor.java:120]: Yes, I believe end goal of fixing the Go AST is to have node lengths in bounds of the input text

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/helm/tree/utils/GoTemplateAstHelper.java (hunk 2 above)
Comment (mstachniuk) [reply iac-extensions/kubernetes/src/main/java/org/sonar/iac/helm/tree/utils/GoTemplateAstHelper.java:58]: finally in Java :-)

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/checks/KubernetesVerifierTest.java [hunk 5]
```diff
@@ -49,7 +49,7 @@ class KubernetesVerifierTest {
 
   @Test
   void verifierShouldSucceedOnHelmProjectWithSecondaryLocationsRaised() {
-    TextRange expectedShiftedTextRangePrimaryIssue = range(7, 0, 7, 40);
+    TextRange expectedShiftedTextRangePrimaryIssue = range(7, 22, 8, 0);
```
Comment (petertrr) [inline iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/checks/KubernetesVerifierTest.java:52]: How reasonable is that the range is shifted onto the next line? Or is it something to be fixed by Go AST adjustments?

**Review by petertrr — APPROVED** [review]

**Pull request merged**